src/Column.tsx

```diff
@@ -22,17 +22,56 @@ class Column extends React.Component<IColumnProps> {
   static FrozenDirection = FrozenDirection;
 }
 
-export interface IColumnProps  {
+
+export interface IColumnEssential {
+  columns?: TColumns;
+  column?: IColumnProps;
+  columnIndex?: number;
+}
+
+export interface IRowEssential<T=any> {
+  rowData?: T[];
+  rowIndex?: number;
+}
+
+export type IColumnCallback<T> = (param: T) => string;
+export type IColumnRenderCallback<T> = (param: T) => React.ReactElement;
+
+export interface IClassNameCBParam<T> extends IColumnEssential, IRowEssential<T> {
+  cellData: any;
+}
+
+export interface IHeaderClassNameCBParam extends IColumnEssential {
+  headerIndex: number;
+}
+
+export interface IDataGetterCBParam<T> extends IColumnEssential, IRowEssential<T> {}
+
+export interface IHeaderRendererCBParam extends IColumnEssential {
+  headerIndex?: number;
+  container?: any;
+  style?: React.CSSProperties;
+}
+
+export interface ICellRendererCBParam<T> extends IColumnEssential, IRowEssential<T> {
+  cellData?: any;
+  container?: any;
+  isScrolling?: boolean;
+  headerIndex?: number;
+  expandIcon?: React.ReactNode;
```

Comment:
ReactNode is a superset of React.Element. 
Hmm. I think expandIcon should be ReactElement (as in other places). However, replacing it is throwing error. Need to discuss in detail.

The error reference (so that I don't forget it):

![image](https://user-images.githubusercontent.com/24398776/63605557-7e098a00-c5ce-11e9-9011-e988cbd9f93b.png)